frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/NPMInstaller.java

```diff
@@ -139,12 +140,28 @@ private void installNpm() throws InstallationException {
                 this.logger.warn("Failed to delete existing NPM installation.");
             }
 
-            extractFile(archive, nodeModulesDirectory);
+            File packageDirectory = new File(nodeModulesDirectory, "package");
+            try {
+                extractFile(archive, nodeModulesDirectory);
+            } catch (ArchiveExtractionException e) {
+                if (e.getCause() instanceof EOFException) {
+                    // https://github.com/eirslett/frontend-maven-plugin/issues/794
+                    // The downloading was probably interrupted and archive file is incomplete:
+                    // delete it to retry from scratch
+                    this.logger.error("The archive file {} is corrupted and will be deleted. "
+                            + "Please try the build again.", archive.getPath());
+                    archive.delete();
+                    if (packageDirectory.exists()) {
+                        FileUtils.deleteDirectory(packageDirectory);
+                    }
+                }
+
+                throw e;
```

Comment:
the possible alternate option here is to re-download one more time
but the probability of this failure is minor, so in case if it happens the user should not even understand what happened :D

Reply:
Related issue - https://github.com/eirslett/frontend-maven-plugin/issues/882 - auto-delete triggers but the auto-retry probably would be nice as a future enhancement

Reply:
Actually, downloader would be benefit from downloading to a tmp location and then moving post download, I put a comment [here](https://github.com/eirslett/frontend-maven-plugin/issues/882#issuecomment-1542390591).